addons/html_editor/static/src/others/embedded_components/core/toggle/toggle.js

```diff
@@ -0,0 +1,39 @@
+import {
+    getEditableDescendants,
+    getEmbeddedProps,
+    useEditableDescendants,
+} from "@html_editor/others/embedded_component_utils";
+import { browser } from "@web/core/browser/browser";
+import { Component, useState } from "@odoo/owl";
+
+const localStorage = browser.localStorage;
+export class EmbeddedToggleComponent extends Component {
+    static template = "html_editor.EmbeddedToggle";
+    static props = {
+        host: { type: Object },
+        toggleId: { type: String },
+    };
+
+    setup() {
+        useEditableDescendants(this.props.host);
+        this.state = useState({
+            showContent: localStorage.getItem(this.toggleStorageKey) === "true",
+        });
+    }
+
+    get toggleStorageKey() {
+        return `Toggle${this.props.toggleId}.showContent`;
```

Comment:
Maybe worth adding "html_editor" at the start of it? So it's more obvious to determine what the key is about when inspecting the local storage.